controllers/clustergroupupgrade_controller.go

```diff
@@ -1241,6 +1266,31 @@ func (r *ClusterGroupUpgradeReconciler) updateStatus(ctx context.Context, cluste
 
 	return nil
 }
+func (r *ClusterGroupUpgradeReconciler) blockingCRsRunning(ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) ([]string, error) {
+
+	var blockingCRs []string
+	r.Log.Info("[blockingCRsRunning]", "len(blockingCRs)", len(blockingCRs))
+
+	for _, blockingCR := range clusterGroupUpgrade.Spec.BlockingCRs {
+		cgu := &ranv1alpha1.ClusterGroupUpgrade{}
+		err := r.Get(ctx, types.NamespacedName{Name: blockingCR.Name, Namespace: blockingCR.Namespace}, cgu)
+
+		if err != nil {
+			r.Log.Info("[blockingCRsRunning] CR not found", "name", blockingCR.Name)
+			return nil, err
+		}
+
+		r.Log.Info("[blockingCRsRunning] condition ", "cgu.Status.Conditions", cgu.Status.Conditions)
+		for i := range cgu.Status.Conditions {
+			if cgu.Status.Conditions[i].Reason == "UpgradeNotCompleted" {
```

Comment:
Does this status exist on an upgrade CR as soon as it is created? If so, this is OK. If this status is only added once the upgrade starts I think the logic needs to be changed to check for any blocking CR that is not completed. I am worried that the user creates CR "y" which is blocked by "x". If "x" has not started yet for some reason and so it does not have "UpgradeNotCompleted" then "y" will start running even though "x" hasn't completed.

Reply:
Well, in that scenario then "y" starts and has its status "UpgradeNotCompleted", then when "x" is ready to start, it won't because it's blocked by "y"'s status. This is, of course, if both "x" and "y" define each other as blocking. This feature ensures "x" and "y" don't run at the same time, but it doesn't mean "x" should run first.